tests/js/object-tests.js

```diff
@@ -515,5 +515,30 @@ module.exports = {
         TestCase.assertEqual(realm.objects('Date')[1].currentDate.getTime(), -10000);
         TestCase.assertEqual(realm.objects('Date')[2].currentDate.getTime(), 1000000000000);
         TestCase.assertEqual(realm.objects('Date')[3].currentDate.getTime(), -1000000000000);
+    },
+    
+    testInheritance: function() {
+        // the equivalent of `class InheritingObject extends Realm.Object`
+        function InheritingObject() {
```

Comment:
It seems like we already have tests for this on React with ES6 classes, courtesy of @appden: https://github.com/realm/realm-js/commit/26d2c169a9adb4f541433c46d02b5351a941c08b.

Too bad we weren't running the React tests on Node 😭